cmd/storj/cmd/cp.go

```diff
@@ -35,76 +37,141 @@ func init() {
 func copy(cmd *cobra.Command, args []string) (err error) {
 	ctx := process.Ctx(cmd)
 
+	identity, err := cpCfg.Load()
+	if err != nil {
+		return err
+	}
+
+	bs, err := cpCfg.GetBucketStore(ctx, identity)
+	if err != nil {
+		return err
+	}
+
 	if len(args) == 0 {
-		return errs.New("No file specified for copy")
+		fmt.Println("No file specified for copy")
+		return nil
 	}
 
 	if len(args) == 1 {
-		return errs.New("No destination specified")
+		fmt.Println("No destination specified")
+		return nil
 	}
 
-	so, err := getStorjObjects(ctx, cpCfg)
+	u0, err := utils.ParseURL(args[0])
 	if err != nil {
 		return err
 	}
 
-	u, err := url.Parse(args[0])
+	u1, err := utils.ParseURL(args[1])
 	if err != nil {
 		return err
 	}
 
-	if u.Scheme == "" {
+	// if uploading
+	if u0.Scheme == "" {
+		if u1.Scheme == "" {
+			fmt.Println("Invalid destination")
+			return nil
+		}
+
+		// if object name not specified, default to filename
+		if u1.Path == "" || u1.Path == "/" {
+			u1.Path = filepath.Base(args[0])
+		}
+
 		f, err := os.Open(args[0])
 		if err != nil {
 			return err
 		}
 
-		fi, err := f.Stat()
+		defer f.Close()
```

Comment:
Shouldn't this be `defer utils.LogClose(f)` to avoid an errcheck error?

Reply:
Done